core/core-api/src/main/java/com/hbc/pms/core/api/service/report/ReportGenerationService.java

```diff
@@ -0,0 +1,111 @@
+package com.hbc.pms.core.api.service.report;
+
+import com.hbc.pms.core.api.config.report.ReportConfiguration;
+import com.hbc.pms.core.api.support.data.ReportExcelProcessor;
+import com.hbc.pms.core.model.Report;
+import io.vavr.control.Try;
+import java.nio.file.Paths;
+import java.util.Collection;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicInteger;
+import lombok.AllArgsConstructor;
+import lombok.SneakyThrows;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.stereotype.Service;
+
+@Service
+@AllArgsConstructor
```

Comment:
"We should use the annotation provided above."
```suggestion
@RequiredArgsConstructor
```

Reply:
Anyway, it looks good for me